src/libraries/System.Console/src/System/Console.cs

```diff
@@ -33,66 +34,80 @@ public static class Console
         private static ConsoleCancelEventHandler? s_cancelCallbacks;
         private static ConsolePal.ControlCHandlerRegistrar? s_registrar;
 
-        internal static T EnsureInitialized<T>([NotNull] ref T? field, Func<T> initializer) where T : class =>
-            LazyInitializer.EnsureInitialized(ref field, ref InternalSyncObject, initializer);
+        internal static T EnsureInitialized<T>([NotNull] ref T? field, T value) where T : class
+            // Interlocked.CompareExchange ensures we don't overwrite a field that was set.
+            => Interlocked.CompareExchange(ref field, value, null) ?? value;
 
-        public static TextReader In => EnsureInitialized(ref s_in, () => ConsolePal.GetOrCreateReader());
+        internal static T EnsureInitializedDisposable<T>([NotNull] ref T? field, T value) where T : class, IDisposable
+        {
+            if (Interlocked.CompareExchange(ref field, value, null) != null)
+            {
+                value.Dispose();
+            }
+
+            return field;
+        }
+
+        public static TextReader In => Volatile.Read(ref s_in) ?? ConsolePal.EnsureInitializedIn(ref s_in);
 
         public static Encoding InputEncoding
         {
             get
             {
-                return EnsureInitialized(ref s_inputEncoding, () => ConsolePal.InputEncoding);
+                return Volatile.Read(ref s_inputEncoding) ?? EnsureInitializedInputEncoding();
+
+                static Encoding EnsureInitializedInputEncoding()
```

Comment:
Is this split into a partial method really needed?

Reply:
I was trying to make sure the JIT would inline this. Should I leave out the static local initializer functions everywhere? Or just for the Encodings?

Reply:
This is not a performance critical method. It is not useful to go out of your way to make it inlineable. In fact, you will see `NoInlining` on number of Console methods to tell the JIT not bother inlining it - it is not worth the extra code size.